app/models/setting.rb

```diff
@@ -198,10 +208,12 @@ def self.convert_array_to_regexp(array)
 
   def self.create_existing(s, opts)
     bypass_readonly(s) do
-      attrs = column_check([:default, :description, :full_name])
+      encrypted = opts.delete(:encrypted)
+      attrs = column_check([:default, :description, :full_name, :encrypted])
       to_update = Hash[opts.select { |k,v| attrs.include? k }]
       to_update.merge!(:value => SETTINGS[opts[:name].to_sym]) if SETTINGS.key?(opts[:name].to_sym)
       s.update_attributes(to_update)
+      s.update_column :encrypted, encrypt if s.encrypted != encrypted
```

Comment:
Tests will confirm, but I think you need to take into account whether the column exists here (which is what column_check is doing above, but you're not using).